lib/WasmReader/WasmBinaryReader.cpp

```diff
@@ -616,40 +611,25 @@ WasmBinaryReader::FunctionBodyHeader()
         switch (type)
         {
         case Wasm::WasmTypes::I32:
-            i32Count += count;
+            m_funcInfo->AddLocal(Wasm::WasmTypes::I32, count);
+            TRACE_WASM_DECODER(L"Function body header: I32 local");
             break;
         case Wasm::WasmTypes::I64:
-            i64Count += count;
+            m_funcInfo->AddLocal(Wasm::WasmTypes::I64, count);
+            TRACE_WASM_DECODER(L"Function body header: I64 local");
```

Comment:
Looking now, this whole switch is kind of pointless now... Should just AddLocal(type, count). Will update in morning with this.